lib/iris/util.py

```diff
@@ -453,7 +453,8 @@ def reverse(cube_or_array, coords_or_dims):
     """
     index = [slice(None, None)] * cube_or_array.ndim
 
-    if iris.cube._is_single_item(coords_or_dims):
+    if (iris.cube._is_single_item(coords_or_dims) or
+            isinstance(coords_or_dims, iris.cube.Cube)):
```

Comment:
Explicitly check for a cube here, on the assumption that the issues in my main comment are specific to cubes.